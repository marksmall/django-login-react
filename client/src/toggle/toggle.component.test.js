import React from 'react';
import { shallow, mount } from 'enzyme';

import Toggle from './toggle.component';
import Switch from './switch.component';

describe('Toggle Component', () => {
  it('should toggle state on', () => {
    const onToggle = jest.fn();

    const testee = mount(
      <Toggle onToggle={() => {}}>
        {({ on, toggle }) => <Switch on={on} onClick={toggle} />}
      </Toggle>
    );

    expect(testee).toMatchSnapshot();
    expect(onToggle.mock.calls.length).toEqual(0);
    expect(testee.find('.toggle-btn').length).toEqual(1);
  });
});
