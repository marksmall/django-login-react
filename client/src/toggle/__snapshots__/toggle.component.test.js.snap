// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Component should toggle state on 1`] = `
<Toggle
  onToggle={[Function]}
>
  <Switch
    on={false}
    onClick={[Function]}
  >
    <div>
      <input
        checked={false}
        className="toggle-input"
        onChange={[Function]}
        type="checkbox"
      />
      <button
        aria-label="Toggle"
        className="toggle-btn
      toggle-btn-off"
        disabled={false}
        onClick={[Function]}
      />
    </div>
  </Switch>
</Toggle>
`;
